### PR TITLE
Changed Search bar style for new UI [1805 - DESKTOP UI]

### LINKED ISSUE #1805: [UI 2.0] Search Page - SearchInput Component
**What would you like to be added**:
Please check #1804 for reference. 
Files can be found here: `src/frontend/next/src/components/SearchInput`

### LINKED ISSUE #1806: [UI 2.0] Search Page - SearchBar Component
**What would you like to be added**:
For reference please check #1804 
File can be found here: `src/frontend/next/src/components/SearchBar.tsx`

### PR DESCRIPTION
**Pull request by huynguyez**
Closes #1805
Closes #1806 

Changed styles for search bar as below:

p

## Type of Change

- [ ] **Bugfix**: Change which fixes an issue
- [ ] **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [x] **UI**: Change which improves UI

## Checklist

- [ ] **Quality**: This PR builds and passes our npm test and works locally
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
- [ ] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

### CODE REVIEW

**Review by PedroFonsecaDEV — APPROVED** [review]
nice!

**Comment by huynguyez** [comment]
> nice!

I would create another issue for mobile UI from this one. Also, I think it's good to keep the search bar sticky on top? what do you think?

**Review by PedroFonsecaDEV — CHANGES_REQUESTED** [review]
@huynguyez 
the responsive for mobile is missing.

**Comment by PedroFonsecaDEV** [comment]
> > nice!
> 
> @PedroFonsecaDEV I would create another issue for mobile UI from this one. Also, I think it's good to keep the search bar sticky on top? What do you think?

I'm confused, 
1) Are you going to file an issue to create the mobile version? 

2) I think if we change to a sticky search bar on top, we will end up with a bulk "Header". We have the scroll to top button to go back to the search bar. But we can think of something like a dropdown search bar, where we can click on it and its popup.

File: src/web/src/components/SearchBar.tsx [hunk 1]
```diff
@@ -20,14 +20,24 @@ const useStyles = makeStyles((theme: Theme) =>
...
+      padding: theme.spacing(0, 0, 0, 2),
+      display: 'flex',
+      flexDirection: 'column',
+      alignItems: 'center',
+      borderRadius: '50px',
+      background: '#d1d1d1',
+      transition: 'background-color .5s',
+      '&:hover': {
+        backgroundColor: '#ffffff',
+        border: 'solid 1px',
```
Comment (humphd) [inline src/web/src/components/SearchBar.tsx:38]: Can we add a `border: solid 1px transparent` when not hovered so it doesn't jump?
outline-bug

**Review by humphd — CHANGES_REQUESTED** [review]
This is looking really cool!

File: src/web/src/components/SearchBar.tsx (hunk 1 above)
Comment (huynguyez) [reply src/web/src/components/SearchBar.tsx]: Done. I will tweak this for mobile responsiveness

**Comment by huynguyez** [comment]
> > > nice!
> > 
> > 
> > @PedroFonsecaDEV I would create another issue for mobile UI from this one. Also, I think it's good to keep the search bar sticky on top? What do you think?
> 
> I'm confused,
> 
> 1. Are you going to file an issue to create the mobile version? (If so, please open the issue, then refer to it here, so we know that we can approve this one)
> 2. I think if we change to a sticky search bar on top, we will end up with a bulk "Header". We have the scroll to top button to go back to the search bar. But we can think of something like a dropdown search bar, where we can click on it and its popup.

Trying to figure out a way to make it looks clean with sticky, probably we won't use sticky title for search page.

**Review by humphd — CHANGES_REQUESTED** [review]
The vertical alignment of the drop-down arrow seems too high

Screen Shot 2021-02-27 at 2 29 29 PM

Also, the relative margins on the text (Post vs. react) in this seem off.

**Comment by PedroFonsecaDEV** [comment]
> > > > nice!
> > > 
> > > 
> > > @PedroFonsecaDEV I would create another issue for mobile UI from this one. Also, I think it's good to keep the search bar sticky on top? What do you think?
> > 
> > 
> > I'm confused,
> > 
> > 1. Are you going to file an issue to create the mobile version? (If so, please open the issue, then refer to it here, so we know that we can approve this one)
> > 2. I think if we change to a sticky search bar on top, we will end up with a bulk "Header". We have the scroll to top button to go back to the search bar. But we can think of something like a dropdown search bar, where we can click on it and its popup.
> 
> Trying to figure out a way to make it looks clean with sticky, probably we shouldn't use sticky title for search page.

@huynguyez 
For now, let's keep the same behaviour as before; using the `backToTop` button and keeping the title on the top of the page.

What about my question 1?

**Comment by huynguyez** [comment]
> The vertical alignment of the drop-down arrow seems too high
> 
> Screen Shot 2021-02-27 at 2 29 29 PM
> 
> Also, the relative margins on the text (Post vs. react) in this seem off.

The dropdown (TextField) can be moved down easily, not sure for the arrow alone
Screen Shot 2021-02-27 at 3 53 56 PM

**Comment by huynguyez** [comment]
> > > > > nice!
> > > > 
> > > > 
> > > > @PedroFonsecaDEV I would create another issue for mobile UI from this one. Also, I think it's good to keep the search bar sticky on top? What do you think?
> > > 
> > > 
> > > I'm confused,
> > > 
> > > 1. Are you going to file an issue to create the mobile version? (If so, please open the issue, then refer to it here, so we know that we can approve this one)
> > > 2. I think if we change to a sticky search bar on top, we will end up with a bulk "Header". We have the scroll to top button to go back to the search bar. But we can think of something like a dropdown search bar, where we can click on it and its popup.
> > 
> > 
> > Trying to figure out a way to make it looks clean with sticky, probably we shouldn't use sticky title for search page.
> 
> @huynguyez
> For now, let's keep the same behaviour as before; using the `backToTop` button and keeping the title on the top of the page.
> 
> What about my question 1?

Just thought I should separate them into 2 issues for desktop & mobile at first, n
ow probly keep them in one.

**Comment by PedroFonsecaDEV** [comment]
> > > > > > nice!
> > > > > 
> > > > > 
> > > > > @PedroFonsecaDEV I would create another issue for mobile UI from this one. Also, I think it's good to keep the search bar sticky on top? What do you think?
> > > > 
> > > > 
> > > > I'm confused,
> > > > 
> > > > 1. Are you going to file an issue to create the mobile version? (If so, please open the issue, then refer to it here, so we know that we can approve this one)
> > > > 2. I think if we change to a sticky search bar on top, we will end up with a bulk "Header". We have the scroll to top button to go back to the search bar. But we can think of something like a dropdown search bar, where we can click on it and its popup.
> > > 
> > > 
> > > Trying to figure out a way to make it looks clean with sticky, probably we shouldn't use sticky title for search page.
> > 
> > 
> > @huynguyez
> > For now, let's keep the same behaviour as before; using the `backToTop` button and keeping the title on the top of the page.
> > What about my question 1?
> 
> Just thought I should separate them into 2 issues for desktop & mobile at first, n
> ow probly keep them in one.

@huynguyez 
Ok. So I will wait for the mobile version on this PR.
Thanks

**Review by birtony — CHANGES_REQUESTED** [review]
Looking good so far, let's get the mobile version in as well 👍🏼

**Comment by PedroFonsecaDEV** [comment]
@huynguyez rebase

**Comment by PedroFonsecaDEV** [comment]
@huynguyez 
Screen Shot 2021-03-07 at 6 14 26 PM

Add an extra break point for this case?

**Comment by huynguyez** [comment]
> @huynguyez
> Screen Shot 2021-03-07 at 6 14 26 PM
> 
> Add an extra break point for this case?

I don't think there's any breakpoint smaller than 'XS'. We should either change that placeholder text to something else or resize it & remove the search icon

**Comment by PedroFonsecaDEV** [comment]
> > @huynguyez
> > Screen Shot 2021-03-07 at 6 14 26 PM
> > Add an extra break point for this case?
> 
> I don't think there's any breakpoint smaller than 'XS'. We should either change that placeholder text to something else or resize it & remove the search icon

You can use numbers instead of letters example: `[theme.breakpoints.down(3000000)]: {}`
But I think a new placeholder would be a good solution too.

Please discuss with the group to find the best solution in the next triage meeting(Tuesday, March 9).

**Comment by humphd** [comment]
@huynguyez this is still not updated from our last meeting, and needs to go in today.  What's happening?

**Comment by izhuravlev** [comment]
@huynguyez Three things: 
1. It looks amazing on desktop, I absolutely love it
2. It looks a bit weird on mobile (Chrome dev tools with iPhone X), and I am talking mostly about the selection part - it, being on top of the search panel and making the search that wide is kinda confusing me: 
image
3. I don't know if it is an issue with the search bar, but it doesn't show any "special characters" - on the screenshot above I sent "Ilya + Gatsby" - and this issue is also related to our old search bar.

**Comment by huynguyez** [comment]
> @huynguyez Three things:
> 
> 1. It looks amazing on desktop, I absolutely love it
> 2. It looks a bit weird on mobile (Chrome dev tools with iPhone X), and I am talking mostly about the selection part - it, being on top of the search panel and making the search that wide is kinda confusing me:
>    image
> 3. I don't know if it is an issue with the search bar, but it doesn't show any "special characters" - on the screenshot above I sent "Ilya + Gatsby" - and this issue is also related to our old search bar.  (cc @humphd @PedroFonsecaDEV )

use ```&``` and it removed all chars from ```&``` to the end. I will file new issues.

**Review by humphd — APPROVED** [review]

**Review by izhuravlev — APPROVED** [review]
LGTM! Good Job, the search bar is just marvelous

**Comment by izhuravlev** [comment]
Question @huynguyez @PedroFonsecaDEV does this PR also resolve #1806?

**Comment by huynguyez** [comment]
> Question to @huynguyez @PedroFonsecaDEV:
> Does this PR also resolve #1806?

@izhuravlev I believe so, updated!

**Pull request merged**